src/openrct2/peep/Guest.cpp

```diff
@@ -4880,55 +4882,131 @@ void Guest::UpdateRideMazePathfinding()
     }
 
     CoordsXY targetLoc = { DestinationX & 0xFFE0, DestinationY & 0xFFE0 };
+    uint16_t stationBaseZ = ride->stations[0].GetBaseZ();
```

Comment:
```suggestion
    uint16_t stationBaseZ = static_cast<uint16_t>(ride->stations[0].GetBaseZ());
```

`GetBaseZ()` is an `int32_t`, implicitly casting to `uint16_t` might be a bad idea. Do you need to remove the signed though? (it was previously `int16_t`)

You only use it for the `map_get_track_element_at` which expects a `CoordsXYZ` object, so I'd say just call this directly there, without casting and such